### PR TITLE
Apply feedback

### PR DESCRIPTION
**Pull request by shanemachine92**
Fixed all of the styling issues, wrote tests, added win_check helper method, and fixed cell overwrite bug

### CODE REVIEW

File: lib/app.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    erb :players
+  end
+
+  get '/game' do
+    session[:game] = Game.new(session[:player1], session[:player2])
+    @game = session[:game]
+    erb :game
+  end
+
+  post '/set_cell' do
```
Comment (cbnoonan) [inline lib/app.rb]: this looks exactly like app.rb  why do we need two files that do the same thing?

File: lib/game.rb [hunk 2]
```diff
@@ -1,81 +1,40 @@
...
-			elsif @total_moves == 9
-				@game_over = true
-				puts 'Neither side is victorious. Maker, help us.'
-			end
-			@current_player = switch_player(@current_player, @player1, @player2)
-		end
-
-		@board.render
-	end
+attr_accessor :player1, :player2, :board, :current_player, :total_moves, :message, :game_over
```
Comment (cbnoonan) [inline lib/game.rb]: probably want to indent this line

File: lib/game.rb [hunk 3]
```diff
@@ -1,81 +1,40 @@
...
+      if @current_player == @player1
+        @player2
+      else
+        @player1
+      end
+    @current_player
+  end
+
+  def play_game(cell, piece)
+      if @board.cell_empty?(cell)
```
Comment (cbnoonan) [inline lib/game.rb]: 4 space indent and then 2 space indent.  You want to have consistency.

**Review by cbnoonan — CHANGES_REQUESTED** [review]

File: lib/app.rb (hunk 1 above)
Comment (shanemachine92) [reply lib/app.rb]: That's weird, lib/app.rb should have been deleted. Dupe file.

File: lib/board.rb [hunk 4]
```diff
@@ -3,56 +3,40 @@ class Board
...
   end
 
-    def board_rows
-    rows = [
-      @cells[:one], @cells[:two], @cells[:three],
-      @cells[:four], @cells[:five], @cells[:six],
-      @cells[:seven], @cells[:eight], @cells[:nine]
-    ]
+  def game_won?(piece)
+    return true if win?(piece)
```
Comment (JesseBuesking) [inline lib/board.rb:21]: Could just be `win?(piece)`

File: lib/board.rb [hunk 5]
```diff
@@ -3,56 +3,40 @@ class Board
...
   end
 
-  def board_columns
-    columns = [
-      @cells[:one], @cells[:four], @cells[:seven],
-      @cells[:two], @cells[:five], @cells[:eight],
-      @cells[:three], @cells[:six], @cells[:nine]
-    ]
+  def win_helper(x, y, z)
+    !(x).empty? && x == y && y == z
```
Comment (JesseBuesking) [inline lib/board.rb:25]: Can probably do `x.present? && x == y && y == z`

File: lib/board.rb [hunk 6]
```diff
@@ -3,56 +3,40 @@ class Board
...
   end
 
-  def board_diagonals
-     diagonals = [
-      @cells[:one], @cells[:five], @cells[:nine],
-      @cells[:three], @cells[:five], @cells[:seven]
-    ]
+  def win?(cell)
+    if win_helper(@cells[:one], @cells[:two], @cells[:three]) ||
```
Comment (JesseBuesking) [inline lib/board.rb:29]: Can just keep the conditional. It will resolve to `true` or `false` itself.

File: lib/game.rb [hunk 7]
```diff
@@ -1,81 +1,40 @@
...
+    @current_player
+  end
+
+  def play_game(cell, piece)
+    if @board.cell_empty?(cell)
+      @board.set_cell(cell, piece)
+      @total_moves += 1
+    end
+
+    if @board.game_won?(piece)
```
Comment (JesseBuesking) [inline lib/game.rb:33]: This could be `@game_over = @total_moves == 9 || @board.game_won?(piece)`

**Review by JesseBuesking — APPROVED** [review]
Just some pointers/comments, no changes required.

**Review by ryanabooth — CHANGES_REQUESTED** [review]
I think the win checker might could use some help:
screen shot 2017-08-08 at 10 04 02 pm

And you can keep playing after a win (hacks!) 😏
not finished yet

**Comment by ryanabooth** [comment]
Sorry I was so late to the party Shane. 😃

File: lib/board.rb (hunk 5 above)
Comment (cbnoonan) [reply lib/board.rb:25]: I would recommend this change. :)

Comment (shanemachine92) [reply lib/board.rb:25]: This change broke the game!

**Comment by shanemachine92** [comment]
@ryanabooth fixed! Good eye, the lost fix for that is in the latest commit.

File: lib/board.rb (hunk 5 above)
Comment (JesseBuesking) [reply lib/board.rb:25]: Why did it break the game?

**Review by JesseBuesking — COMMENTED** [review]
Left a question inline.

File: lib/board.rb (hunk 5 above)
Comment (shanemachine92) [reply lib/board.rb:25]: @JesseBuesking 
def win_helper(x, y, z)
  x.present? && x == y && y == z
 end

resulted in the following error:

NoMethodError - undefined method `present?' for "M":String
Did you mean?  prepend:
	/Users/shane/work/JSON_MvT/lib/board.rb:25:in `win_helper'
	/Users/shane/work/JSON_MvT/lib/board.rb:29:in `win?'
	/Users/shane/work/JSON_MvT/lib/board.rb:21:in `game_won?'
	/Users/shane/work/JSON_MvT/lib/game.rb:32:in `play_game'
	app.rb:29:in `block in <main>

Comment (shanemachine92) [reply lib/board.rb:25]: It's a rails method and I'm using Sinatra. Tried installing and requiring active support, but still breaking. 

https://stackoverflow.com/questions/18810562/how-to-use-ruby-method-present

I can keep trying if it feels like an important change.

Comment (cbnoonan) [reply lib/board.rb:25]: it's not a hugely important change, but it is good to know how to use bits and pieces from Rails if you need to. I'm curious what error you received when trying to include ActiveSupport?

Comment (shanemachine92) [reply lib/board.rb:25]: Same error as above :/

**Pull request closed without merging**